zivid_camera/src/capture_settings_controller.cpp

```diff
@@ -10,6 +10,33 @@
 
 #include <dynamic_reconfigure/server.h>
 
+namespace
+{
+template <typename Node>
+void recursivelyFillInUnsetWithCameraDefault(Node& node, const Zivid::CameraInfo& cameraInfo)
+{
+  if constexpr (Node::nodeType == Zivid::DataModel::NodeType::group ||
+                Node::nodeType == Zivid::DataModel::NodeType::leafDataModelList)
+  {
+    node.forEach([&cameraInfo](auto& child) { recursivelyFillInUnsetWithCameraDefault(child, cameraInfo); });
+  }
+  else if (!node.hasValue())
+  {
+    static_assert(Node::nodeType == Zivid::DataModel::NodeType::leafValue);
+    node = Zivid::Experimental::SettingsInfo::defaultValue<Node>(cameraInfo);
+  }
+}
+
+template <typename ZividSettingsType>
+ZividSettingsType fillInUnsetWithCameraDefault(const ZividSettingsType& settings, const Zivid::CameraInfo& cameraInfo)
+{
+  ZividSettingsType copy{ settings };
```

Comment:
Why not pass `settings` by-value if you anyway making a copy?

Reply:
Yeah could do that, but I think it doesn't matter much here, since at the call site I would anyway make the copy. Currently there is 1 copy, line 33. Passing by value there would be 1 copy and 1 move.
 I could also take settings by reference and modify it, but found returning a copy cleaner.